scapy/layers/inet.py

```diff
@@ -931,7 +932,8 @@ def mysummary(self):
 bind_layers(CookedLinux, IP, proto=2048)
 bind_layers(GRE, IP, proto=2048)
 bind_layers(SNAP, IP, code=2048)
-bind_layers(Loopback, IP, type=0)
+if not consts.OPENBSD:
+    bind_layers(Loopback, IP, type=0)
```

Comment:
This works for OpenBSD, but possibly not for other OS:

bind_layers(Loopback, IPv6, type=0x18)
bind_bottom_up(Loopback, IPv6, type=0x1c)
bind_bottom_up(Loopback, IPv6, type=0x1e)

What is your idea?  Register all with bind_bottom_up() and only one per OS with bind_top_down()?  I doubt that will help reading foreign pcap files from loopback with OpenBSD or vice versa as the encoding of the address family is still different.

Reply:
I’m not a big fan of having OS-specific bindings hanging around in the layer files :/

Even though it probably won’t be used, I’d rather have the code you posted above (it will work for other platforms)

Reply:
Or we could also use `bind_layers(Loopback, IPv6, type=socket.AF_INET6)` I guess

Reply:
I like the idea of type=socket.AF_INET6.  For consistency I also did this for IPv4.  To bind the unspecified type to IPv4 I have used bind_bottom_up.  The new commit works for OpenBSD and should work for other OS.